src/main/java/seedu/mark/logic/parser/TabCommandParser.java

```diff
@@ -0,0 +1,30 @@
+package seedu.mark.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.mark.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.mark.commons.core.index.Index;
+import seedu.mark.logic.commands.TabCommand;
+import seedu.mark.logic.parser.exceptions.ParseException;
+
+public class TabCommandParser implements Parser<TabCommand> {
+
+    public static final String MESSAGE_INVALID_INDEX = "Index is neither 1, 2 nor 3.";
+
+    public TabCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(args);
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TabCommand.MESSAGE_USAGE), pe);
+        }
+
+        if (index.getOneBased() < 0 || index.getOneBased() > 0) {
+            ParseException pe = new ParseException(MESSAGE_INVALID_INDEX);
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TabCommand.MESSAGE_USAGE), pe);
+        }
```

Comment:
Would this fit in the `TabCommand` class? To mirror the way `EditCommand` works. 
Also, should the first ParseException `pe` be thrown so that the user gets to see `MESSAGE_INVALID_INDEX`?

[Edit: Or if we convert `Tab` to an enum, we may not need this check at all.]

Reply:
Okay I'll move it to `TabCommand`, but will need this check still (to convert to enum types). 
For the exception I'll do away with the 2nd exception (i.e. throw the `MESSAGE_INVALID_INDEX`) since the `MESSAGE_INVALID_COMMAND_FORMAT` does not seem very useful.

Reply:
Another thing: would it be better to let the user type out the name of the tab instead of index? (name feels more intuitive to me, though I need to type more)
eg.
`tab dashboard` to switch to dashboard

(not urgent)

Reply:
That sounds useful! We can explore it in future commits. And maybe also provide shorter keywords like `tab dash`, `tab on`, and `tab off`?